tv_apps/notification-receiver/_test/unit/receiver_test.js

```diff
@@ -0,0 +1,99 @@
+/* global MockPresentation, MockPresentationSession, MockNotification,
+          Receiver */
+
+'use strict';
+
+require('/js/receiver.js');
+require('/shared/test/unit/mocks/mock_notification.js');
+require('/tv_apps/tv_shared/test/unit/mocks/mock_presentation.js');
+require('/tv_apps/tv_shared/test/unit/mocks/mock_presentation_session.js');
+
+suite('notification-receiver/Receiver', function() {
+  var receiver;
+  var realPresentation;
+  var realNotification;
+
+  suiteSetup(function() {
+    realPresentation = window.navigator.presentation;
+    window.navigator.presentation = MockPresentation;
+
+    realNotification = window.Notification;
+    window.Notification = MockNotification;
+  });
+
+  suiteTeardown(function() {
+    window.navigator.presentation = realPresentation;
+    window.Notification = realNotification;
+  });
+
+  suite('init()', function() {
+    setup(function() {
+      this.sinon.spy(MockPresentationSession, 'addEventListener');
+      receiver = new Receiver();
+    });
+
+    teardown(function() {
+      MockPresentationSession.addEventListener.restore();
+      navigator.presentation._mReset();
+      receiver.uninit();
+    });
+
+    test('> _onSessionReady should called once ' +
+      'when session object is not ready', function() {
+        this.sinon.spy(receiver, '_sessionReadyHandler');
+
+        assert.isFalse(receiver._sessionReadyHandler.called);
+        receiver.init();
+        navigator.presentation._mInjectSession(
+          MockPresentationSession._mCreateSession());
+        assert.isTrue(receiver._sessionReadyHandler.calledOnce);
```

Comment:
I'm not sure if we should let private property being involved in testing.

But since I got no answer if we should forbid this kind of assertion for now, I'll just leave a comment here. Personally I'd prefer not to test private property.